packages/core/src/sync-store/postgres/migrations.ts

```diff
@@ -457,28 +457,65 @@ const migrations: Record<string, Migration> = {
     },
   },
   "2024_03_20_0_checkpoint_in_logs_table": {
+    async up(_db: Kysely<any>) {
+      // no-op migration to avoid crashing databases that successfully ran this migration
+      return;
+    },
+  },
+  "2024_04_04_0_log_events_indexes": {
     async up(db: Kysely<any>) {
+      await db.schema.dropIndex("blockNumberIndex").ifExists().execute();
+      await db.schema.dropIndex("blockTimestampIndex").ifExists().execute();
+
+      await db.schema
+        .createIndex("logBlockNumberIndex")
+        .on("logs")
+        .column("blockNumber")
+        .execute();
+    },
+  },
+  "2024_04_11_0_add_checkpoint_column_to_logs_table": {
+    async up(db: Kysely<any>) {
+      if (await hasDoneCheckpointMigration(db)) {
+        return;
+      }
```

Comment:
I wonder if it would be better to use `ADD COLUMN IF NOT EXISTS` rather than the manual check (https://stackoverflow.com/a/38721951). Also, I think now we could remove this migration, and do the `ADD COLUMN IF NOT EXISTS` to the "2024_03_20_0_checkpoint_in_logs_table" migration above.

Reply:
oh! i didnt know this was a thing! i actually tried to do this, but i didnt see it in the kyseley API, so i thought it wasn't a thing at all. i'll change it to use raw SQL. good catch

Reply:
<img width="494" alt="Screen Shot 2024-04-14 at 5 33 48 PM" src="https://github.com/ponder-sh/ponder/assets/3744812/46b62d67-59a2-44aa-b10d-7fbfdc16407d">
no `ifNotExists` for `addColumn` in kysely. but i think doing the raw SQL is worth it for simplifying the sql